src/test/spec/unified_runner/mod.rs

```diff
@@ -165,40 +165,41 @@ pub async fn run_unified_format_test(test_file: TestFile) {
                         }
                     }
 
-                    if let Some(ref expect_result) = operation.expect_result {
-                        let result = result
-                            .unwrap_or_else(|e| {
-                                panic!(
-                                    "{} should succeed, but the following error: {}",
-                                    operation.name, e
-                                )
-                            })
-                            .unwrap_or_else(|| {
+                    if let Some(expect_error) = operation.expect_error {
```

Comment:
The only change here is refactoring the conditionals to unwrap the result if `expect_error` is not present rather than just when `expect_result` is present.